contracts/Proposal.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity ^0.5.8;
+
+import "./zeppelin/token/ERC20/IERC20.sol";
+import "./zeppelin/token/ERC20/SafeERC20.sol";
+import "./ownership/Ownable.sol";
+import "./Basket.sol";
+
+/**
+ * A Proposal represents a suggestion to change the backing for RSV.
+ *
+ * The lifecycle of a proposal:
+ * 1. Creation
+ * 2. Acceptance
+ * 3. Completion
+ *
+ * A time can be set during acceptance to determine when completion is eligible.
+ * A proposal can also be closed before it is Completed. 
+ *
+ * This contract is intended to be used in one of two possible ways. Either:
+ * - A target RSV basket is suggested, and quantities to be exchanged are  
+ *     deduced at the time of proposal execution.
+ * - A specific quantity of tokens to be exchanged is suggested, and the 
+ *     resultant RSV basket is determined at the time of proposal execution.
+ *
+ */
+contract Proposal is Ownable {
```

Comment:
I have some inclination for this to be a struct rather than a full contract. I need to think through it more, though.

Did you start with this being a struct instead?

Reply:
Yep, started as a struct. I forget the precise reason, but the struct approach didn't play nice with having array subfields.

Reply:
Actually, maybe it was Basket as a subfield that was a problem. I forget.